zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -113,26 +116,132 @@ private ZeebeRocksDbFactory(
 
   /** @return Options which are used on all column families */
   public ColumnFamilyOptions createColumnFamilyOptions() {
-    // start with some defaults
-    final var columnFamilyOptionProps = new Properties();
-    // look for cf_options.h to find available keys
-    // look for options_helper.cc to find available values
-    columnFamilyOptionProps.put("compaction_pri", "kOldestSmallestSeqFirst");
+    final var userProvidedColumnFamilyOptions = rocksDbConfiguration.getColumnFamilyOptions();
+    final var hasUserOptions = !userProvidedColumnFamilyOptions.isEmpty();
+
+    if (hasUserOptions) {
+      return createFromUserOptions(userProvidedColumnFamilyOptions);
+    }
+
+    return createDefaultColumnFamilyOptions();
+  }
 
-    // apply custom options
-    columnFamilyOptionProps.putAll(userProvidedColumnFamilyOptions);
+  private ColumnFamilyOptions createDefaultColumnFamilyOptions() {
+    final var columnFamilyOptions = new ColumnFamilyOptions();
+    final var tableConfig = createTableFormatConfig();
+
+    columnFamilyOptions
+        .setCompactionPriority(CompactionPriority.OldestSmallestSeqFirst)
+        // compaction
+        .setLevelCompactionDynamicLevelBytes(true)
+        .setCompactionPriority(CompactionPriority.OldestSmallestSeqFirst)
+        .setCompactionStyle(CompactionStyle.LEVEL)
+        // configure 4 levels: L1 = 32mb, L2 = 320mb, L3 = 3.2Gb, L4 >= 3.2Gb
```

Comment:
All the L0 settings are missing - I understand that we used to base the values of off the memtables, which we don't configure here, but we should still define them. This will help us have a bit more control on the compaction that gets triggered for L0->L1. Considering memtables by default are 64MB, we could just do it for every second memtable? We could omit the stop writes just to be safe.

```java
    // L-0 means immediately flushed memtables
        .setLevel0FileNumCompactionTrigger(2)
        .setLevel0SlowdownWritesTrigger(3)
```

Reply:
Was not sure about this. The default value for `Level0FileNumCompactionTrigger` is 4 which I thought is already small enough. So was not sure if there is any advantage in configuring it.

Reply:
You can leave it for now then - I'm also not 100% sure what it means concretely. I think a smaller value will trigger compaction of L0 to L1 faster, which I hope would help with fragmentation, but I'm also not sure. 4 * 64MB seems like a lot potentially (if we assume L0 files can be as big as a single memtable), but I'm also not 100% sure about it.